cmd2/parsing.py

```diff
@@ -25,93 +28,76 @@ class Statement(str):
 
     The string portion of the class contains the arguments, but not the command, nor
     the output redirection clauses.
+    """
+    # the arguments, but not the command, nor the output redirection clauses.
+    args = attr.ib(default='', validator=attr.validators.instance_of(str), type=str)
 
-    :var raw:               string containing exactly what we input by the user
-    :type raw:              str
-    :var command:           the command, i.e. the first whitespace delimited word
-    :type command:          str or None
-    :var multiline_command: if the command is a multiline command, the name of the
-                            command, otherwise None
-    :type command:          str or None
-    :var args:              the arguments to the command, not including any output
-                            redirection or terminators. quoted arguments remain
-                            quoted.
-    :type args:             str or None
-    :var: argv:             a list of arguments a la sys.argv. Quotes, if any, are removed
-                            from the elements of the list, and aliases and shortcuts
-                            are expanded
-    :type argv:             list
-    :var terminator:        the character which terminated the multiline command, if
-                            there was one
-    :type terminator:       str or None
-    :var suffix:            characters appearing after the terminator but before output
-                            redirection, if any
-    :type suffix:           str or None
-    :var pipe_to:           if output was piped to a shell command, the shell command
-                            as a list of tokens
-    :type pipe_to:          list
-    :var output:            if output was redirected, the redirection token, i.e. '>>'
-    :type output:           str or None
-    :var output_to:         if output was redirected, the destination file
-    :type output_to:        str or None
+    # string containing exactly what we input by the user
+    raw = attr.ib(default='', validator=attr.validators.instance_of(str), type=str)
```

Comment:
In a perfect world, I would make nearly all of these keyword-only arguments; however ``attrs`` didn't add the **kw_only** argument until the most recent version (18.2.0) which just got released on Sept 1, 2018.  I'd rather not force a required 3rd-party dependency on something that has only been available for about a week because this could make it inconvenient to use ``cmd2`` on older platforms.  However, we may want to consider refactoring this a year or so down the road.